### DISCUSSION TITLE
Legal usage of Microsoft Marketplace Extensions

### DISCUSSION THREAD

**Discussion by J-Rios** [coder/code-server#6026] [General]
Hello,

I really love **code-server**, however it is not completely clear for me if I can install and use extensions that are not available in [Open VSX Registry](https://open-vsx.org/), and are only available on [Microsoft Marketplace](https://marketplace.visualstudio.com/vscode) due it Terms of use.

From [Microsoft Marketplace Terms of Use](https://cdn.vsassets.io/v/M146_20190123.39/_content/Microsoft-Visual-Studio-Marketplace-Terms-of-Use.pdf):

```text
Visual Studio Products and Services. Your right to use any underlying Visual Studio Products
and Services will be governed by the agreement under which you purchased such products or services,
and will be subject to the payment of fees for such products or services, where applicable. Marketplace
Offerings are intended for use only with Visual Studio Products and Services and you may only install
and use Marketplace Offerings with Visual Studio Products and Services.
```

It is clear for me that the Microsoft Marketplace can't be used by non-Microsoft tools (like code-server), but not sure if the above Right also apply to the use of the extensions already published on that Marketplace.

I have seen some user comments that recomend to just get and install extensions using *.vsix files...

My question are:

1. It is still a violation of Microsoft terms of use if we get a *.vsix file and install it?

2. How about getting *.vsix files from outside Microsoft Marketplace (i.e. from an extension github repository) and it license allow it usage, it is fine to get and install it in code-server this way?

Best Regards!

**Comment by code-asher** [comment]
I am not a lawyer but my understanding is that installing .vsix 
files hosted on Microsoft's marketplace into code-server (or 
VSCodium, etc) is a violation of the terms regardless of the 
acquisition method (manually downloaded from the website, 
automated through the API, etc).

If the .vsix is in the GitHub repository and the license allows 
usage I believe it is fair game to download and install it into 
code-server.  In this case it should also qualify to be added on 
Open VSX if it is not already there.
